fili-core/src/main/java/com/yahoo/bard/webservice/data/metric/protocol/BuiltInMetricProtocols.java

```diff
@@ -0,0 +1,88 @@
+// Copyright 2020 Oath Inc.
+// Licensed under the terms of the Apache license. Please see LICENSE.md file distributed with this work for terms.
+package com.yahoo.bard.webservice.data.metric.protocol;
+
+import java.util.HashSet;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * A collection of signal names supported.
+ */
+public class BuiltInMetricProtocols {
+
+    // The names for default protocols
+    // Intentionally mutable
+    private static final Set<String> DEFAULT_PROTOCOL_CONTRACTS = new HashSet<>();
+
+    // The dictionary for default protocols
+    // intentionally mutable
+    private static final ProtocolDictionary PROTOCOL_DICTIONARY = new ProtocolDictionary();
```

Comment:
Why do have both of these fields? Why not just PROTOCOL_DICTIONARY, with DEFAULT_PROTOCOL_CONTRACTS
derived from PROTOCOL_DICTIONARY.keys()? Doesn't having two collections risk error, like adding
a protocol name to DEFAULT_PROTOCOL_CONTRACTS, but forgetting to add the protocol to PROTOCOL_DICTIONARY
(or vice versa) ?

Reply:
Oh wait, I think I might understand now. You don't need to (and shouldn't) always add new
protocols `DEFAULT_PROTOCOL_CONTRACTS` and `PROTOCOL_DICTIONARY`. You only add a protocol name to
`DEFAULT_PROTOCOL_CONTRACTS` if you want *every metric in your system* to support it by default. If you add
the Protocol to just `PROTOCOL_DICTIONARY`, you register it with the system, but you have to explicitly
configure metrics to handle it.